### PR TITLE
Added pragma option to suppress warnings for relations.

### PR DESCRIPTION
**Pull request by azreika**
<b> Gets rid of the "Warning: No rules/facts defined for..." for given relations when toggled. </b>

Usage:
```
.pragma "suppress" "LIST"
```
where LIST is a comma-separated list of relations to suppress warnings for, with no spaces.

E.g.
```
.pragma "suppress" "a,c"

.decl a(x:number)
.decl b(x:number)
.decl c(x:number)
.output c()
```
will only produce the warning

```
Warning: No rules/facts defined for relation b in file sup.dl at line 4
.decl b(x:number)
------^-----------
```
instead of
```
Warning: No rules/facts defined for relation a in file sup.dl at line 2
.decl a(x:number)
------^-----------
Warning: No rules/facts defined for relation b in file sup.dl at line 3
.decl b(x:number)
------^-----------
Warning: No rules/facts defined for relation c in file sup.dl at line 4
.decl c(x:number)
------^-----------
```
<b> Fixes a bug with pragma options on the command line. </b>
Previously, pragma options on the command line were delimited with a comma. This conflicts with the fact that some pragma option values are also delimited with commas (e.g. MST, this PR, etc.). Changed the delimiter to be a semicolon (;) now.

E.g.:
```souffle -D- whatever.dl --pragma="magic-transform:myrelation,myotherelation;suppress:anotherone,relationA"```

### CODE REVIEW

**Comment by souffle-lang** [comment]
[65% (-0.04%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/215/)

**Comment by b-scholz** [comment]
Can you add an example for the suppress pragma? 
I don't like "suppress" on its own.
Also, could we change "suppress"  to "suppress-warning" or "no-warning" ?

File: src/AstSemanticChecker.cpp [hunk 1]
```diff
@@ -636,7 +656,7 @@ void AstSemanticChecker::checkRelation(ErrorReport& report, const TypeEnvironmen
     }
 
     // check whether this relation is empty
-    if (relation.clauseSize() == 0 && !relation.isInput()) {
+    if (relation.clauseSize() == 0 && !relation.isInput() && !relation.isSuppressed()) {
```
[the remark this answers is not in the document]
Comment (azreika) [reply src/AstSemanticChecker.cpp]: From what I've seen, nullaries do usually have non-trivial rules defining them, so I don't like the idea of automatically disabling warnings for them, especially if we can suppress them using the new option.

**Comment by azreika** [comment]
> Can you add an example for the suppress pragma?
> I don't like "suppress" on its own.
> Also, could we change "suppress" to "suppress-warning" or "no-warning" ?

Yep, good point! I'll change it to `suppress-warning` and throw in an example.

**Comment by souffle-lang** [comment]
[65% (0.0%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/220/)

**Comment by souffle-lang** [comment]
[65% (-0.02%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/221/)

**Comment by souffle-lang** [comment]
[65% (+0.01%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/222/)

**Pull request merged**